From: Miklos Vajna <vmiklos@frugalware.org>
Date: Sat, 3 May 2008 01:13:33 +0000 (+0200)
Subject: fwnet_setlastprofile(): create lockfile with proper permissions
X-Git-Url: http://git.frugalware.org/gitweb/gitweb.cgi?p=frugalwareutils.git;a=commitdiff_plain;h=e6103e355926c062e8b18218115a07c1117ee846

	fwnet_setlastprofile(): create lockfile with proper permissions
	- in case umask would be something weird
	- closes #3052
---

diff --git a/libfwnetconfig/libfwnetconfig.c b/libfwnetconfig/libfwnetconfig.c
index 8134e60..75da63d 100644
--- a/libfwnetconfig/libfwnetconfig.c
+++ b/libfwnetconfig/libfwnetconfig.c
@@ -568,6 +568,7 @@ char *fwnet_lastprofile(void)
  */
 int fwnet_setlastprofile(char* str)
 {
+	int fd;
 	FILE *fp;
 
 	if(!str)
@@ -578,7 +579,8 @@ int fwnet_setlastprofile(char* str)
 			return(0);
 	}
 
-	fp = fopen(FWNET_LOCK, "w");
+	fd = creat(FWNET_LOCK, 0644);
+	fp = fdopen(fd, "w");
 	if(fp==NULL)
 		return(1);
 	fprintf(fp, str);
diff --git a/libfwnetconfig/Makefile b/libfwnetconfig/Makefile
index a420a9c..83c9a59 100644
--- a/libfwnetconfig/Makefile
+++ b/libfwnetconfig/Makefile
@@ -23,7 +23,7 @@ LDFLAGS += $(shell pkg-config --libs glib-2.0)
 
 # fwutil
 CFLAGS += -I../libfwutil
-LDFLAGS += -L../libfwutil -lfwutil
+LDFLAGS += -L../libfwutil -lfwutil -lresolv
 
 #SUBDIRS = po
 LIBNAME = netconfig
